sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/LimitRequestBodySizeFilter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.security;
+
+import java.io.IOException;
+import java.util.Enumeration;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.util.unit.DataSize;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+// based on this hint: https://stackoverflow.com/a/38611154
+@Component
+public class LimitRequestBodySizeFilter extends OncePerRequestFilter {
+
+  private final DataSize requestBodyMaxSize;
+
+  public LimitRequestBodySizeFilter(
+      @Value("${request.body.max.size}") DataSize requestBodyMaxSize) {
+    this.requestBodyMaxSize = requestBodyMaxSize;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+    if (request.getContentLengthLong() > requestBodyMaxSize.toBytes()) {
+      response.sendError(HttpStatus.PAYLOAD_TOO_LARGE.value()); // 413
+    } else if (isChunkedEncoding(request.getHeaders(HttpHeaders.TRANSFER_ENCODING))) {
+      response.sendError(HttpStatus.LENGTH_REQUIRED.value()); // 411
+    } else {
```

Comment:
Wouldn't it be easier to add a test for a missing CONTENT_LENGTH header? Or we could add it as an additional condition.

Reply:
That would break all GET, HEAD and OPTIONS requests and most (close to all) DELETE requests. But the RFC does not strictly forbid payload for these requests. So it would lead to a lot more conditions to separate the methods. It felt just less complex and less prone to errors to verify the chunked encoding in general.